src/Storage/Query/Handler/SelectQueryHandler.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt\Storage\Query\Handler;
+
+use Bolt\Entity\Content;
+use Bolt\Storage\Query\ContentQueryParser;
+use Bolt\Storage\Query\QueryResultset;
+use Bolt\Storage\Query\SelectQuery;
+
+/**
+ *  Handler class to perform select query and return a resultset.
+ */
+class SelectQueryHandler
+{
+    /**
+     * @return Content[]|Content|false
+     */
+    public function __invoke(ContentQueryParser $contentQuery)
+    {
+        $set = new QueryResultset();
+        /** @var SelectQuery $query */
+        $query = $contentQuery->getService('select');
+        $query->setSingleFetchMode(false);
+
+        foreach ($contentQuery->getContentTypes() as $contentType) {
+            $contentType = str_replace('-', '_', $contentType);
+
+            $repo = $contentQuery->getContentRepository();
```

Comment:
is there a single content repo for all content types? then it should be constructed inside getService('select'):
inside SelectQuery:
private setQueryBuilder()
private setContentType() (setAlias?) (here also the where part from https://github.com/bolt/four/pull/116/files#diff-5ee7c092d66585260210d467fb97c416R50)